src/helpers/script_helpers.sh

```diff
@@ -117,6 +145,15 @@ function run {
     preconditions "$@"
     steps > "$STEPS_FILE"
     run_steps "$STEPS_FILE" undoable
+
+    # Apply global previous-branch restoration
+    # only if we're at the git root.
+    # Any branch restoration logic for the other
+    # case needs to be handled in a custom way by
+    # the command itself
```

Comment:
I'm addressing something that has not been tested, which is that restoring the previous branch works as expected after aborting / continuing when they are conflicts. Unsure if we can actually handle aborting correctly with our current architecture, but continuing would be possible if the steps for resetting the previous branch were added to the steps file.

I see two ways of accomplishing this.

1) We go back to your original approach of updating each command with custom logic. This allows nice optimization for specific commands but does muddy each commands logic.

2) We add steps to the steps file after the command has written its basic steps. To avoid the extra stash push and pop, it might make sense to also pull these out of each command's logic and into the architecture. I'm thinking of something like
```sh
# src/helpers/script_helpers.sh
function run {
  if [ "$1" = "--abort" ]; then
    ensure_abortable
    abort_command
  elif [ "$1" = "--continue" ]; then
    ensure_continuable
    ensure_no_conflicts
    continue_command
  elif [ "$1" = "--skip" ]; then
    ensure_skippable
    skip_command
  elif [ "$1" = "--undo" ]; then
    ensure_undoable
    undo_command
  else
    remove_step_files
    preconditions "$@"
    pre_steps > "$STEPS_FILE" # This adds the logic for stashing 
    steps >> "$STEPS_FILE"
    post_steps >> "$STEPS_FILE" # This adds the logic for restoring the previous branch and then popping the stash
    run_steps "$STEPS_FILE" undoable
  fi
}
```
Each command may have some configuration like should stash and what are the possible previous branches to set to (if they exist)